### PR TITLE
Disable xtest 20000 (FS corruption tests) when OP-TEE is build for RPMB

### PR DESCRIPTION
**Pull request by jforissier**
When the filesystem is implemented over RPMB (CFG_RPMB_FS=y), there is
not point trying to corrupt files in the REE.

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
Would it be possible to do something similar when using emulated RPMB?

**Comment by jforissier** [comment]
@jenswi-linaro yes it should be possible to do something similar, but it would not be straightforward because:
1. The tests we have currently look very specific to the REE implementation, so we would need to rewrite them
2. File corruption cannot be done with a simple fopen/fwrite. We would need a communication channel to the tee-supplicant, which does not exist currently.

**Comment by jenswi-linaro** [comment]
@jforissier OK
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by pascal-brand-st-dev** [comment]
I would prefer to keep coherency between what we have in conf.mk, and what is defined. This means leave CFG_ENC_FS defined if it is 'y' in conf.mk. I would prefer to fix .c, from

    #ifdef CFG_ENC_FS

into

    #if defined(CFG_ENC_FS) && !defined(CFG_RPMB_FS)

**Comment by jforissier** [comment]
@pascal-brand-st-dev you're right, it's cleaner. Updated.

File: host/xtest/xtest_20000.c [hunk 1]
```diff
@@ -11,7 +11,7 @@
  * GNU General Public License for more details.
  */
 
-#ifdef CFG_ENC_FS
+#if defined(CFG_ENC_FS) && !defined(CFG_RPMB_FS)
```
Comment (pascal-brand-st-dev) [inline host/xtest/xtest_20000.c]: Not needed in `xtest_main.c` and `xtest_test.h`?

Comment (jforissier) [inline host/xtest/xtest_20000.c]: Sorry forgot to add them. Fixed.

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jforissier** [comment]
Rebased and tags applied.

**Pull request merged**